src/network/fragment_array_input_stream.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include <initializer_list>
+#include <vector>
+
+#include "network/fragment_input_stream.h"
+
+namespace lightstep {
+/**
+ * A non-owning stream of fragments.
+ */
+class FragmentArrayInputStream final : public FragmentInputStream {
+ public:
+  FragmentArrayInputStream() noexcept = default;
+
+  explicit FragmentArrayInputStream(std::initializer_list<Fragment> fragments);
+
+  FragmentArrayInputStream& operator=(
+      std::initializer_list<Fragment> fragments);
+
+  // FragmentInputStream
```

Comment:
This [SpanStream](https://github.com/rnburn/lightstep-tracer-cpp/blob/stream-header/src/recorder/stream_recorder/span_stream.h) class that wraps the CircularBuffer to expose it as fragments will also implement the interface and have different logic for the Seek function since it needs to track spans that are partially written.